hieunc-guestbook-dojo/static/js/guestbook/store/GreetingStore.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Created by hieunc on 27/11/2015.
+ */
+
+define([
+	"dojo/_base/declare",
+	"dojo/_base/lang",
+	"dojo/Deferred",
+	"dojo/request/xhr",
+	"dojo/cookie"
+], function (declare, lang, Deferred, xhr, cookie) {
+	var widgetGuestbookList = declare("guestbook.store.GreetingStore", null, {
+		url: "/api/guestbook/default_guestbook/greeting/",
+		headers: {},
+		method: "get", //get,post,put,del
+		postData: {},
+		handleAs: "json",
+		expect: "", //{json,httpStatus}
+
+		callBack: function (result) {
+		},
+		errCallBack: function (err) {
+			console.log(err)
+		},
+
+		constructor: function (/*Object*/ kwArgs) {
+			lang.mixin(this, kwArgs);
+		},
+
+		_getResult: function () {
+			var _self = this;
+			var _def = new Deferred();
+			_def.then(this.callBack, this.errCallBack);
+
+			xhr[this.method](_self.url, {
+				//sync: true,
+				headers: _self.headers,
+				preventCache: _self.preventCache,
+				data: _self.postData,
+				handleAs: _self.handleAs
+			}).then(
+					function (data) {
```

Comment:
- hãy dùng:
```
lang.hitch(this, function(data) {
	if (this.expect === "json") {
		_def.resolve(data);
	}
})
```
- trong trường hợp xảy ra điều kiện `this.expect !== "json"` thì điều gì sẽ xảy ra?